build/patches/Timezone-Override.patch

```diff
@@ -0,0 +1,1379 @@
++const char kContentSettingsCustomTimezone[] = "profile.content_settings.custom_timezone";
++
+ }  // namespace prefs
+diff --git a/components/content_settings/core/common/pref_names.h b/components/content_settings/core/common/pref_names.h
+--- a/components/content_settings/core/common/pref_names.h
++++ b/components/content_settings/core/common/pref_names.h
+@@ -55,6 +55,8 @@ extern const char kManagedWebUsbAskForUrls[];
+ extern const char kManagedWebUsbBlockedForUrls[];
+ extern const char kManagedLegacyCookieAccessAllowedForDomains[];
+ 
++extern const char kContentSettingsCustomTimezone[];
++
+ }  // namespace prefs
+ 
+ #endif  // COMPONENTS_CONTENT_SETTINGS_CORE_COMMON_PREF_NAMES_H_
+diff --git a/components/content_settings/renderer/content_settings_agent_impl.cc b/components/content_settings/renderer/content_settings_agent_impl.cc
+--- a/components/content_settings/renderer/content_settings_agent_impl.cc
++++ b/components/content_settings/renderer/content_settings_agent_impl.cc
+@@ -38,6 +38,11 @@
+ #include "url/gurl.h"
+ #include "url/origin.h"
+ #include "url/url_constants.h"
++#include "third_party/blink/renderer/core/timezone/timezone_controller.h"
++#include "third_party/blink/renderer/core/inspector/locale_controller.h"
++#include "third_party/blink/renderer/core/inspector/locale_controller.h"
++#include "third_party/icu/source/common/unicode/strenum.h"
++#include "third_party/icu/source/i18n/unicode/timezone.h"
+ 
+ using blink::WebDocument;
+ using blink::WebFrame;
+@@ -48,6 +53,8 @@ using blink::WebURL;
+ using blink::WebView;
+ using content::DocumentState;
+ 
++std::unique_ptr<blink::TimeZoneController::TimeZoneOverride> timezone_override_;
++
+ namespace content_settings {
+ namespace {
+ 
+@@ -346,6 +353,9 @@ bool ContentSettingsAgentImpl::AllowScript(bool enabled_per_settings) {
+   allow = allow || IsWhitelistedForContentSettings();
+ 
+   cached_script_permissions_[frame] = allow;
++  
++  if (allow) UpdateOverrides();
++
+   return allow;
+ }
+ 
+@@ -572,4 +582,104 @@ bool ContentSettingsAgentImpl::AllowStorageAccess(
+   return result;
+ }
+ 
++bool ContentSettingsAgentImpl::UpdateOverrides() {
++  // Evaluate the content setting rules
++  ContentSetting setting = CONTENT_SETTING_ALLOW;
++
++  if (content_setting_rules_) {
++    blink::WebLocalFrame* frame = render_frame()->GetWebFrame();
++
++    setting = GetContentSettingFromRules(
++        content_setting_rules_->timezone_override_rules, frame,
++        url::Origin(frame->GetDocument().GetSecurityOrigin()).GetURL());
++  }
++  return UpdateTimeZoneOverride(setting, content_setting_rules_->timezone_override_value);
++         //&& UpdateLocaleOverride(setting);
++}
++
++// bromite #331: Customize Timezone
++bool ContentSettingsAgentImpl::UpdateTimeZoneOverride(ContentSetting setting, const std::string& timezone_override_value) {
++  // base/i18n/icu_util.cc # 329
++  //VLOG(0) << "---Bromite:ContentSettingsAgentImpl::UpdateTimeZoneOverride start";
```

Comment:
You can leave them uncommented since they will be active only when debugging.

See also: https://groups.google.com/a/chromium.org/forum/#!msg/chromium-dev/3NDNd1KzXeY/AZKMMx37fdQJ

Reply:
from what I've tried, `LOG(INFO)` is deleted from the non-debug version, `VLOG(0)` is kept instead.
I used `VLOG()` because `#define LOG(INFO)` is not always present as opposed to `VLOG()`, which is always available

Reply:
Then `LOG(DEBUG)` should be used here; you can add the include for it (`"base/logging.h"`) where missing.

Reply:
>  ("base/logging.h") where missing.

no, it's not always enough, even if I still don't understand why.
for now I follow your advice, then I will check.

Reply:
```
../../components/content_settings/renderer/content_settings_agent_impl.cc:604:3: error: use of undeclared identifier 'COMPACT_GOOGLE_LOG_DEBUG'
  LOG(DEBUG) << "---Bromite:ContentSettingsAgentImpl::UpdateTimeZoneOverride start";
```

as I told you it does not work.
i think it is related to `LOG()` needs some configuration somewhere.
since nothing changes, I leave the `VLOG`s as 1

Reply:
That's because `LOG(DEBUG)` is invalid, you should use `LOG(INFO)`; I was mistaken in my example.

Reply:
you're right so it works, I don't know what I did